server/src/main/java/org/opensearch/telemetry/tracing/channels/TraceableHttpChannel.java

```diff
@@ -49,7 +48,7 @@ private TraceableHttpChannel(HttpChannel delegate, Span span, Tracer tracer) {
      * @return http channel
      */
     public static HttpChannel create(HttpChannel delegate, Span span, Tracer tracer) {
-        if (FeatureFlags.isEnabled(FeatureFlags.TELEMETRY) == true) {
+        if (tracer.isRecording()) {
```

Comment:
Super nit (here and other places), to follow conventions:

```suggestion
        if (tracer.isRecording() == true) {
```